src/sedcli_main.c

```diff
@@ -654,6 +701,41 @@ static int handle_setpw(void)
 	return ret;
 }
 
+static int handle_mbr_control(void)
+{
+	struct sed_device *dev = NULL;
+	int ret;
+
+	sedcli_printf(LOG_INFO, "Enter Admin1 password: ");
```

Comment:
this should go after sedcli_init to improve user experience.
Now if user specifies invalid disk he is prompted for password and after entering password sedcli errors out because of invalid disk. All necessary checks (including discovery and check if drive is in proper state)  should be done before prompting for password to improve user experience

Reply:
Will do it